src/gui/uithememanager.cpp

```diff
@@ -49,16 +107,21 @@ void UIThemeManager::freeInstance()
 
 void UIThemeManager::initInstance()
 {
-    if (!m_instance)
+    if (!m_instance) {
         m_instance = new UIThemeManager;
+        loadCustomPalette();
```

Comment:
Why is it here?
IMO, UIThemeManager constructor is better place.

Reply:
this uses `UIThemeManager::instance()->getColor()`, if done in constructor, `UIThemeManager::instance()` will return invalid value, should I change it to `loadCustomPalette(UIThemeManager *)` then it can be called from constructor?

Reply:
Why not to make it class private method `applyCustomPalette() `?

There are two ways:
1. UIThemeManager manages this logic (then apply proposed changes)
2. Application manages this logic (then move `loadCustomPalette() ` in Application class and call after UIThemeManager instance is created).

Reply:
> * UIThemeManager manages this logic (then apply proposed changes)
> * Application manages this logic (then move `loadCustomPalette() ` in Application class and call after UIThemeManager instance is created).

what do you prefer?
against 1, I can think that UIThemeManager should do only one thing, so It shouldn't do the palette thingy (idk I just started reading Uncle Bob),

Reply:
> against 1, I can think that UIThemeManager should do only one thing, so It shouldn't do the palette thingy (idk I just started reading Uncle Bob),


well it already applies stylesheet so these argument doesn't hold much, will do 1st way as it will require fewer changes.

Reply:
Well, let's use second one. I only object to have this code outside of affected classes.

Reply:
>will do 1st way since it require fewer changes.

Ok.